update boot tests documentation and scripts

aakahlow opened the pull request:
All the updated configuration files in this PR are from gem5-resources/boot-tests, so there is no need to review them here. The main document to be reviewed is: docs/tutorials/boot-tutorial.md

aakahlow committed:
update boot tests documentation and scripts

powerjg commented on docs/gem5-configs/configs-boot-tests/run_exit.py:
```diff
@@ -30,48 +30,54 @@
 """
```
I notice there's an empty file documentation string here :)

powerjg commented on docs/gem5-configs/configs-boot-tests/run_exit.py:
```diff
@@ -83,12 +89,14 @@
     # instantiate all of the objects we've created above
     m5.instantiate()
 
+    globalStart = time.time()
```
Is this an unused variable?

powerjg reviewed (commented):
Is there any way to reference gem5-resources instead of having two copies of python files to keep in sync?

Can you point to where this is on readthedocs so we can read it more easily?

aakahlow committed:
remove boot tests config files from here

aakahlow commented:
> Is there any way to reference gem5-resources instead of having two copies of python files to keep in sync?

Yeah, that's a much better idea. I have removed config files from this repo and changed the links in tutorial to gem5-resources configs.

> Can you point to where this is on readthedocs so we can read it more easily?

I have triggered a build for this branch on readthedocs. Once built, it should be available on: https://gem5art.readthedocs.io/en/boot-tests-docs/

powerjg commented on docs/tutorials/boot-tutorial.md:
```diff
@@ -75,7 +91,7 @@ See the commands below:
 git clone https://gem5.googlesource.com/public/gem5
 cd gem5
-git checkout d40f0bc579fb8b10da7181
+git checkout 90a6e809629c978c2df765eb
```
Is it possible to use a tag here (e.g., v20.1.0.2)?

aakahlow replied:
updated.

powerjg commented on docs/tutorials/boot-tutorial.md:
```diff
@@ -88,32 +104,33 @@ gem5_repo = Artifact.registerArtifact(
 gem5_binary = Artifact.registerArtifact(
     command = '''cd gem5;
-    git checkout d40f0bc579fb8b10da7181;
+    git checkout 90a6e809629c978c2df765ebb70;
```
Ditto, can we use a human-readable tag?

aakahlow replied:
updated

powerjg commented on docs/tutorials/boot-tutorial.md:
```diff
@@ -144,43 +161,68 @@ Now, to build the disk image, inside the disk-image folder, run:
-Let's use an example of kernel v5.2.3 to see how to compile the kernel.
+
+Let's use an example of kernel v5.4.49 to see how to compile the kernel.
 First, add a folder linux-configs to store linux kernel config files. The configuration files of interest are available [here](https://github.com/darchr/gem5art/blob/master/docs/linux-configs/).
```
Did you mean to update this with a gem5-resources pointer?

aakahlow replied:
I have updated the link to gem5-resources linux configs. I will remove linux-configs folder from gem5art/docs once other tutorials are updated as well.

powerjg commented on docs/tutorials/boot-tutorial.md:
```diff
@@ -144,43 +161,68 @@ Now, to build the disk image, inside the disk-image folder, run:
 Next, we need to add gem5 run scripts. We will do that in a folder named configs-boot-tests.
-Get the run script named run_exit.py from [here](https://github.com/darchr/gem5art/blob/master/docs/gem5-configs/configs-boot-tests/run_exit.py), and other system configuration files from
-[here](https://github.com/darchr/gem5art/blob/master/docs/gem5-configs/configs-boot-tests/system/).
+Get the run script named run_exit.py from [here](https://gem5.googlesource.com/public/gem5-resources/+/refs/heads/stable/src/boot-exit/configs/run_exit.py), and other system configuration files from
```
@BobbyRBruce, do you think we should have a v20.1 branch on gem5-resources? We could have that branch last forever and have stable move to point to the same place as the "current version" branch.

This might be more overhead than it's worth, but it would make this documentation more forward-looking.

The other option is to use a specific changeset, but I don't love that solution, either.

powerjg commented on docs/tutorials/boot-tutorial.md:
```diff
@@ -282,35 +329,62 @@ disk_image = Artifact.registerArtifact(
 gem5_binary = Artifact.registerArtifact(
     command = '''cd gem5;
-    git checkout d40f0bc579fb8b10da7181;
+    git checkout 90a6e809629c978c2df765ebb70;
```
Human readable tag?

aakahlow replied:
done

powerjg commented on docs/tutorials/boot-tutorial.md:
```diff
@@ -282,35 +329,62 @@ disk_image = Artifact.registerArtifact(
+
+gem5_binary_MESI_Two_Level = Artifact.registerArtifact(
+    command = '''cd gem5;
+    git checkout 90a6e809629c978c2df765ebb70;
```
Human-readable tag?

aakahlow replied:
done

powerjg commented on docs/tutorials/boot-tutorial.md:
```diff
@@ -282,35 +329,62 @@ disk_image = Artifact.registerArtifact(
+
+gem5_binary_MOESI_CMP_directory = Artifact.registerArtifact(
+    command = '''cd gem5;
+    git checkout 90a6e809629c978c2df765ebb70;
```
Human-readable tag?

aakahlow replied:
done

powerjg commented on docs/tutorials/boot-tutorial.md:
```diff
@@ -325,38 +399,107 @@ linux_binaries = {
+    runs = starmap(createRun, product(linuxes, boot_types, cpu_types, num_cpus, mem_types))
+    # Run all of these experiments in parallel
+    for run in runs:
+        run_gem5_instance.apply_async((run, os.getcwd(),))
```
This line is the only difference between celery and multiprocessing, right? Most of this looks the same.

No need to copy the code twice.

You can move "createRun" out of the `__name__=="__main__"` statement to make it more modular.

aakahlow replied:
Fixed.

powerjg reviewed (commented):
Mostly small comments, and most are the same comment repeated :). Overall, this looks great! Thanks, Ayaz!

aakahlow committed:
small fixes
